bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/RtspHandler.java

```diff
@@ -0,0 +1,178 @@
+import org.eclipse.jdt.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.netty.bootstrap.Bootstrap;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelDuplexHandler;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.channel.socket.nio.NioSocketChannel;
+import io.netty.handler.codec.http.DefaultHttpRequest;
+import io.netty.handler.codec.http.HttpContent;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.rtsp.RtspDecoder;
+import io.netty.handler.codec.rtsp.RtspEncoder;
+import io.netty.handler.codec.rtsp.RtspHeaderNames;
+import io.netty.handler.codec.rtsp.RtspMethods;
+import io.netty.handler.codec.rtsp.RtspVersions;
+import io.netty.util.CharsetUtil;
+
+/**
+ * The {@link RtspHandler} This class is a WIP and is currently not used. Can be used to check what format the stream is
+ * in
+ * mjpeg or H.264/5, check audio formats and frame rates.
+ *
+ *
+ * @author Matthew Skinner - Initial contribution
+ */
+
+@NonNullByDefault
+public class RtspHandler extends ChannelDuplexHandler {
+    @Nullable
+    private Bootstrap rtspBootstrap;
+    private EventLoopGroup mainEventLoopGroup = new NioEventLoopGroup();
+    private String ipAddress = "todo";
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    RtspHandler() {
+        // todo may need to pass in IP of camera plus a few others to make this code
+        // work again after it was moved to its own class.
+    }
+
+    public HttpRequest getRTSPoptions(String rtspURL) {
+        HttpRequest request = new DefaultHttpRequest(RtspVersions.RTSP_1_0, RtspMethods.OPTIONS, rtspURL);
+        request.headers().add(RtspHeaderNames.CSEQ, "1");
+        return request;
+    }
+
+    public HttpRequest getRTSPdescribe(String rtspURL) {
+        HttpRequest request = new DefaultHttpRequest(RtspVersions.RTSP_1_0, RtspMethods.DESCRIBE, rtspURL);
+        request.headers().add(RtspHeaderNames.CSEQ, "2");
+        return request;
+    }
+
+    public HttpRequest getRTSPsetup(String rtspURL) {
+        HttpRequest request = new DefaultHttpRequest(RtspVersions.RTSP_1_0, RtspMethods.SETUP, rtspURL);
+        request.headers().add(RtspHeaderNames.CSEQ, "3");
+        request.headers().add(RtspHeaderNames.TRANSPORT, "RTP/AVP;unicast;client_port=5000-5001");
+        return request;
+    }
+
+    public HttpRequest getRTSPplay(String rtspURL) {
+        HttpRequest request = new DefaultHttpRequest(RtspVersions.RTSP_1_0, RtspMethods.PLAY, rtspURL);
+        request.headers().add(RtspHeaderNames.CSEQ, "4");
+        request.headers().add(RtspHeaderNames.SESSION, "12345678"); // need session number to match that of setup
```

Comment:
Yes it should be random. This class is not used currently and is an area that I am developing as I have spare time. Its probably 80% done to where it can be used. Eventually it will be able to make automatic choices inside the binding based on what format the cameras streams are in, making life easier for people that don't understand the inner workings of cameras.

Reply:
Reworked the class to a new one and made it closer to being ready to being used.